File: dawii_20222_miercoles_backend_grupo_04-master/src/main/java/com/cibertec/service/ProveedorService.java
package com.cibertec.service;

import java.util.List;

import com.cibertec.entity.Proveedor;

public interface ProveedorService {
	
	//Consulta
	public abstract List<Proveedor> listaProveedorPorRazonRucEstadoPais(String razonsocial, String ruc, int estado, int idPais);

	
	//CRUD	
	public abstract Proveedor insertaActualizaProveedor(Proveedor obj);	
	public abstract List<Proveedor> listaProveedorPorRazonLike(String razonsocial);
	public abstract List<Proveedor> listaProveedor(); 
	public abstract void eliminaProveedor(int idProveedor);
}

File: dawii_20222_miercoles_frontend_grupo_04-master/src/app/services/proveedor.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AppSettings } from '../app.settings';
import { Proveedor } from '../models/proveedor.model';

const baseUrlProveedor = AppSettings.API_ENDPOINT + '/proveedor';
const baseUrlProveedorCRUD = AppSettings.API_ENDPOINT + '/crudProveedor';

@Injectable({
  providedIn: 'root'
})
export class ProveedorService {

  constructor(private http: HttpClient) { }

  registraProveedor(obj: Proveedor): Observable<any> {
    return this.http.post(baseUrlProveedor, obj);
  }

  listaProveedor(razonsocial: string, ruc: string, estado: number, idPais: number): Observable<any> {
    const params = new HttpParams().set("razonsocial", razonsocial).set("ruc", ruc).set("estado", estado).set("idPais", idPais);
    return this.http.get(baseUrlProveedor + "/listaProveedorConParametros", { params });
  }

  /* MÉTODOS DEL CRUD */

  consultaProveedor(filtro: string): Observable<Proveedor[]> {
    return this.http.get<Proveedor[]>(baseUrlProveedorCRUD + "/listaProveedorPorRazonLike/" + filtro);
  }

  registraProveedorCrud(obj: Proveedor): Observable<any> {
    return this.http.post(baseUrlProveedorCRUD + "/registraProveedor", obj);
  }

  actualizaProveedor(obj: Proveedor): Observable<any> {
    return this.http.put(baseUrlProveedorCRUD + "/actualizaProveedor", obj);
  }

  eliminarProveedor(idProveedor: number): Observable<any> {
    return this.http.delete(baseUrlProveedorCRUD + "/eliminaProveedor/" + idProveedor);
  }

}
